### PR TITLE
#40 ios file picker

### PR DESCRIPTION
**Pull request by Dorofeev**

### CODE REVIEW

File: media/src/iosMain/kotlin/dev/icerock/moko/media/picker/DocumentPickerDelegateToContinuation.kt [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package dev.icerock.moko.media.picker
```
Comment (Alex009) [inline media/src/iosMain/kotlin/dev/icerock/moko/media/picker/DocumentPickerDelegateToContinuation.kt:1]: copyright required

File: media/src/iosMain/kotlin/dev/icerock/moko/media/picker/ios/MediaPickerController.kt [hunk 2]
```diff
@@ -65,7 +68,21 @@ class MediaPickerController(
     }
 
     override suspend fun pickFiles(): FileMedia {
-        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
+        var delegatePtr: DocumentPickerDelegateToContinuation? // strong reference to delegate (view controller have weak ref)
+        val fileMedia = suspendCoroutine<FileMedia> { continuation ->
+            val localDelegatePtr = DocumentPickerDelegateToContinuation(continuation)
+            delegatePtr = localDelegatePtr
+
+            val controller = UIDocumentPickerViewController(forOpeningContentTypes = listOf(UTType.typeWithIdentifier("public.data")))
```
Comment (Alex009) [inline media/src/iosMain/kotlin/dev/icerock/moko/media/picker/ios/MediaPickerController.kt:76]: what this type means? 
also can we use constant for type identifier? as like `kUTTypeVideo` above.
also maybe we should pass type as argument? or this type means all file types at all?

File: media/src/iosX64Main/kotlin/dev/icerock/moko/media/picker/DocumentPickerDelegateToContinuation.kt [hunk 3]
```diff
@@ -0,0 +1,33 @@
+package dev.icerock.moko.media.picker
```
Comment (Alex009) [inline media/src/iosX64Main/kotlin/dev/icerock/moko/media/picker/DocumentPickerDelegateToContinuation.kt:1]: copyright required

**Review by Alex009 — CHANGES_REQUESTED** [review]

File: media/src/iosMain/kotlin/dev/icerock/moko/media/picker/ios/MediaPickerController.kt (hunk 2 above)
Comment (Dorofeev) [reply media/src/iosMain/kotlin/dev/icerock/moko/media/picker/ios/MediaPickerController.kt:76]: public.data is a parent for all standard types https://developer.apple.com/library/archive/documentation/FileManagement/Conceptual/understanding_utis/art/conformance_hierarchy.gif

File: media/src/iosMain/kotlin/dev/icerock/moko/media/picker/ios/MediaPickerController.kt [hunk 4]
```diff
@@ -65,7 +68,24 @@ class MediaPickerController(
...
 
     override suspend fun pickFiles(): FileMedia {
-        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
+        var delegatePtr: DocumentPickerDelegateToContinuation? // strong reference to delegate (view controller have weak ref)
+        val fileMedia = suspendCoroutine<FileMedia> { continuation ->
+            val localDelegatePtr = DocumentPickerDelegateToContinuation(continuation)
+            delegatePtr = localDelegatePtr
+
+            val controller = UIDocumentPickerViewController(
+                    documentTypes = listOf(kStandardFileTypesId),
```
Comment (Alex009) [inline media/src/iosMain/kotlin/dev/icerock/moko/media/picker/ios/MediaPickerController.kt:77]: as i see in docs - we can use constant `kUTTypeData`

**Review by Alex009 — CHANGES_REQUESTED** [review]

**Review by Alex009 — APPROVED** [review]

**Pull request merged**